server/src/main/java/org/opensearch/indices/NodeIndicesStats.java

```diff
@@ -74,26 +77,27 @@
  */
 @PublicApi(since = "1.0.0")
 public class NodeIndicesStats implements Writeable, ToXContentFragment {
-    private CommonStats stats;
-    private Map<Index, List<IndexShardStats>> statsByShard;
+    protected CommonStats stats;
+    protected Map<Index, CommonStats> statsByIndex;
+    protected Map<Index, List<IndexShardStats>> statsByShard;
 
     public NodeIndicesStats(StreamInput in) throws IOException {
         stats = new CommonStats(in);
-        if (in.readBoolean()) {
-            int entries = in.readVInt();
-            statsByShard = new HashMap<>();
-            for (int i = 0; i < entries; i++) {
-                Index index = new Index(in);
-                int indexShardListSize = in.readVInt();
-                List<IndexShardStats> indexShardStats = new ArrayList<>(indexShardListSize);
-                for (int j = 0; j < indexShardListSize; j++) {
-                    indexShardStats.add(new IndexShardStats(in));
-                }
-                statsByShard.put(index, indexShardStats);
+        if (in.getVersion().onOrAfter(Version.V_3_0_0)) {
```

Comment:
can we make all the newly added references to `Version.V_3_0_0` as `Version.CURRENT`? This will help in backports.

Reply:
I will be raising separate PRs for 2.x and main branch after this PR is merged.

Reply:
Why do you need separate PRs. Have this as CURRENT, CURRENT In main will point to 3.0 and 2.17 in 2.x. The backport of this PR will lead to successful builds at both places. Then create a PR on main moving the version to 2.17 branch and backport the same to 2.x. This is how typically I have done version related changes to avoid any build failure.

Reply:
I see. This is efficient, have raised a commit with `Version.CURRENT`

Reply:
using Version.CURRENT sounds risky to me in case you miss updating. I would rather prefer failing the build then silent issues.